### PR TITLE
feat: Add flag for skip manifest update

### PR DESCRIPTION
**Pull request by yngvark**
# Description

Don't update package manifest if new flag is set.

```sh
go run ../.. pkg update --disable-manifest-update
Updating schema configuration files: out/app-hello, out/app-common, out/networking, out/databases
Transforming configuration to add support for Apex domain routing. File: config/app-hello.yml

✅ Successfully updated packages:
  - out/app-hello                                 app                                      [config/common-config.yml config/app-hello.yml]                                 
  - out/app-common                                app-common                               [config/common-config.yml config/app-common.yml]                                
  - out/networking                                networking                               [config/common-config.yml config/networking.yml]                                
  - out/databases                                 databases                                [config/common-config.yml config/databases.yml]          

$ git st
 M config/app-common.yml
 M config/app-hello.yml
 M config/databases.yml
 M config/networking.yml

$ ls config/.schemas
app-common-v3.5.0.schema.json  app-v9.5.0.schema.json  databases-v3.3.0.schema.json  networking-v2.4.0.schema.json
```

Notice there are no changes to `packages.yml`.

There is a number of changes in this PR. The essential one is `if !opts.DisableManifestUpdate {`. The rest is small quality improvements that should improve ability for humans to understand what's going on. See inline comments.

# Motivation

In our [update infra workflow](https://github.com/oslokommune/pirates-iac/blob/238e8e76fa06cb2859ad42165e45d84bafc16e43/.github/workflows/update-infrastructure-from-golden-path.yml), Renovate has already updated `packages.yml`. So `ok update` must not do it, but still do the rest of what `update` should do.

### CODE REVIEW

File: pkg/pkg/update/update.go [hunk 1]
```diff
@@ -23,28 +24,31 @@ func Run(pkgManifestFilename string, packages []common.Package, opts Options) er
...
-		return fmt.Errorf("saving package manifest: %w", err)
+		err = common.SavePackageManifest(pkgManifestFilename, manifest)
+		if err != nil {
+			return fmt.Errorf("saving package manifest: %w", err)
+		}
 	}
 
 	if opts.UpdateSchemaConfig {
-		err = updateSchemaConfiguration(context.Background(), updatedPackages, manifest, latestReleases)
+		err = updateSchemaConfiguration(context.Background(), packages, manifest, latestReleases)
```
Comment (yngvark) [inline pkg/pkg/update/update.go:44]: Using `updatedPackages` is an an optimisation that increases coupling. This coupling introduces complexity now as we have a new flag. My suggestion is to remove this coupling, by simply doing the same for updating schema config as for the two other operations in this function (updating packages and migrating config), which is to use `packages`.

With this update, the surrounding function (`Run`) does three operations, all operating on `packages`.

File: pkg/pkg/update/update.go [hunk 2]
```diff
@@ -23,28 +24,31 @@ func Run(pkgManifestFilename string, packages []common.Package, opts Options) er
...
-	if err != nil {
-		return fmt.Errorf("saving package manifest: %w", err)
+		err = common.SavePackageManifest(pkgManifestFilename, manifest)
+		if err != nil {
+			return fmt.Errorf("saving package manifest: %w", err)
+		}
 	}
 
 	if opts.UpdateSchemaConfig {
-		err = updateSchemaConfiguration(context.Background(), updatedPackages, manifest, latestReleases)
```
Comment (yngvark) [inline pkg/pkg/update/update.go:40]: This is an optimisation that avoids updating schema config for packages that hasn't been updated. I've moved this optimisation to `manifest.Packages[i].Ref == newRef` (see function `updateSchemaConfiguration`).

File: pkg/pkg/update/update.go [hunk 3]
```diff
@@ -23,78 +24,83 @@ func Run(pkgManifestFilename string, packages []common.Package, opts Options) er
...
 		if !ok {
-			return nil, fmt.Errorf("no latest release found for package: %s", manifestPkg.Template)
+			return common.PackageManifest{}, fmt.Errorf("no latest release found for package: %s", pkg.Template)
 		}
 
-		newRef := fmt.Sprintf("%s-%s", manifestPkg.Template, latestRelease) // e.g. app-v2.1.3
+		newRef := fmt.Sprintf("%s-%s", pkg.Template, latestRelease) // e.g. app-v2.1.3
 
-		if manifestPkg.Ref != newRef {
-			manifestPkg.Ref = newRef
```
Comment (yngvark) [inline pkg/pkg/update/update.go:79]: Replaced code that modified state, with code that returns a new object instead (see also `Clone()` above).

**Comment by staticaland** [comment]
Seems like a good idea to me 👍 Maybe later migration will be its own command

**Review by staticaland — APPROVED** [review]

**Pull request merged**